### PR TITLE
NYS-64: MCP role should not have access to all committees

### PR DESCRIPTION
**Pull request by joelsteidl**
This was created in response to https://atendesign.atlassian.net/browse/NYS-64

Per @routinet 's direction, I added a couple new methods to web/modules/custom/nys_users/src/UsersHelper.php

Totally open to naming convention there. Naming stuff is hard!!!

I ended up using a `hook_query_TAG_alter` because it will be a more flexible solution if we need to further limit access to taxonomy terms for MCP users, etc.

### CODE REVIEW

File: web/modules/custom/nys_users/src/UsersHelper.php [hunk 1]
```diff
@@ -168,4 +168,40 @@ public static function isLcOrMcp(mixed $user): bool {
       );
   }
 
+  /**
+   * Gets a list of all the senators assigned to a user for management.
+   *
+   * @param \Drupal\user\Entity\User|int|null $user
+   *   Either a User entity or the ID of one.  If NULL, current user is used.
+   */
+  public static function getUserSenatorManagement(mixed $user): array {
```
Comment (routinet) [inline web/modules/custom/nys_users/src/UsersHelper.php:177]: rename to `getManagedSenators()`

File: web/modules/custom/nys_users/src/UsersHelper.php [hunk 2]
```diff
@@ -168,4 +168,40 @@ public static function isLcOrMcp(mixed $user): bool {
...
+    return array_column($senators, 'target_id', 'target_id');
+  }
+
+  /**
+   * Gets a list of committees by senator tids found in field_chair.
+   *
+   * @param array $tids
+   *   Senator term IDs.
+   */
+  public static function getCommitteesBySenators(array $tids): array {
```
Comment (routinet) [inline web/modules/custom/nys_users/src/UsersHelper.php:196]: Move to `nys_senators\SenatorsHelper::getChairedCommittees()`

File: web/modules/custom/nys_senators/nys_senators.module [hunk 3]
```diff
@@ -1237,6 +1239,57 @@ function nys_senators_query_nys_senators_access_alter(AlterableInterface $query)
...
+  $tables = $query->getTables();
+  $target_tables = ['taxonomy_term_data', 'taxonomy_term_field_data'];
+  foreach ($tables as $info) {
+    $table = $info['table'];
+    if (!in_array($table, $target_tables)) {
+      continue;
+    }
+    $primary_table = $info['alias'];
+  }
+  // Get the senator management TIDs from the current user.
```
Comment (routinet) [inline web/modules/custom/nys_senators/nys_senators.module:1277]: Create a wrapper method as `nys_users\UsersHelper::getManagedCommittees()` encompassing current lines 1278-1280

**Review by routinet — COMMENTED** [review]
I'm also going to look into how this would be done as a custom access check for an entity query.

**Review by aheaphy — APPROVED** [review]

**Pull request merged**